### PR TITLE
Add dockerfile

### PR DESCRIPTION
**Pull request by chrieke**
As intended in https://github.com/chrieke/prettymapp/pull/42 adds Dockerfile & readme instrunctions.

But keeps the same code folder structure.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,19 @@
+FROM python:3.11-slim
+
+WORKDIR /app
+
+RUN apt-get update && apt-get install -y \
```
Comment (Saibamen) [inline Dockerfile:5]:
```suggestion
RUN apt-get update && apt-get install --no-install-recommends -y \
```

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    curl gdal-bin libgdal-dev g++ \
+    && rm -rf /var/lib/apt/lists/*
```
Comment (Saibamen) [inline Dockerfile:7]:
```suggestion
    && rm -rf /var/lib/apt/lists/* \
    && apt-get autoremove -y \
    && apt-get clean
```

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+RUN mkdir -p /app/streamlit-prettymapp
```
Comment (Saibamen) [inline Dockerfile:9]: This is not needed - will be created during `COPY` command

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+COPY streamlit-prettymapp /app/streamlit-prettymapp
+
+RUN pip3 install -r streamlit-prettymapp/requirements.txt
```
Comment (Saibamen) [inline Dockerfile:13]:
```suggestion
RUN pip3 install --upgrade pip --no-cache-dir \
    && pip3 install -r streamlit-prettymapp/requirements.txt --no-cache-dir
```

File: README.md [hunk 5]
```diff
@@ -72,3 +81,5 @@ fig.savefig("map.jpg")
 To customize the map appearance, use the additional arguments of the [`Plot`](plotting.py#L36) class (e.g. `shape`, 
 `contour_width` etc.). Check the preconfigured [styles](prettymapp/settings.py#L35) and 
 webapp [examples](streamlit-prettymapp/examples.json) for inspiration.
+
+
```
Comment (Saibamen) [inline README.md:85]: Remove double empty lines

File: README.md [hunk 6]
```diff
@@ -38,6 +38,15 @@ pip install -r streamlit-prettymapp/requirements.txt
 streamlit run streamlit-prettymapp/app.py
 ```
 
+## Run the app with Docker
+
+Build and run the Docker container with the existing Docker file. Open `http://0.0.0.0:8501/` in your browser to 
+see the streamlit app.
+```bash
```
Comment (Saibamen) [inline README.md:45]: Fix markdownlint
```suggestion
see the streamlit app.

```bash
```

File: Dockerfile [hunk 7]
```diff
@@ -0,0 +1,19 @@
+FROM python:3.11-slim
```
Comment (Saibamen) [inline Dockerfile:1]: Use latest 3.12 if is possible

**Review by Saibamen — CHANGES_REQUESTED** [review]
Added cleanup suggestions to reduce final docker image size, based on https://github.com/google-deepmind/alphafold/ repo

**Comment by chrieke** [comment]
Thank you @Saibamen!

**Comment by Saibamen** [comment]
>pushes it down to 1.57GB

How much it was before?

File: README.md (hunk 5 above)
Comment (Saibamen) [reply README.md:85]: Not resolved

File: Dockerfile (hunk 7 above)
Comment (Saibamen) [reply Dockerfile:1]: @chrieke ?

File: README.md (hunk 6 above)
Comment (Saibamen) [reply README.md:45]: Not resolved
@chrieke https://github.com/DavidAnson/markdownlint

File: Dockerfile (hunk 3 above)
Comment (Saibamen) [reply Dockerfile:11]: Did you test it @chrieke?

**Review by Saibamen — CHANGES_REQUESTED** [review]

**Comment by chrieke** [comment]
Closing, Dockerfile added in https://github.com/chrieke/prettymapp/pull/42

**Pull request closed without merging**